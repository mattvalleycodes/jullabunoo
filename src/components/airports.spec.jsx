import * as React from "react";
import renderer from "react-test-renderer";

import Airports from "./airports";

describe("<Airports /> component", () => {
  test("renders all of the provided airport information", () => {
    const data = [
      {
        code: "YRI",
        name: "Riviere Du Loup",
        city: "Riviere Du Loup",
        country: "Canada",
        region: "Americas",
        timezone: "America/Montreal",
      },
      {
        code: "CKG",
        name: "Chongqing",
        city: "Chonqing",
        country: "China",
        region: "Asia",
        timezone: "Asia/Shanghai",
      },
    ];

    const airports = renderer.create(<Airports airports={data} />).toJSON();

    expect(airports).toMatchSnapshot();
  });
});
