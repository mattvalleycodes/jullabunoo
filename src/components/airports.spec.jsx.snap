// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Airports /> component renders all of the provided airport information 1`] = `
<ul
  className="airports"
>
  <li
    className="airports__summary"
  >
    <div
      className="box"
    >
      <a
        className="airport-summary"
        href="/airport/YRI"
        onClick={[Function]}
      >
        <h1
          className="airport-summary__name"
        >
          Riviere Du Loup
        </h1>
        <span
          className="airport-summary__code"
        >
          YRI
        </span>
        <span
          className="airport-summary__location"
        >
          Riviere Du Loup
          , 
          Canada
           -- 
          Americas
        </span>
      </a>
    </div>
  </li>
  <li
    className="airports__summary"
  >
    <div
      className="box"
    >
      <a
        className="airport-summary"
        href="/airport/CKG"
        onClick={[Function]}
      >
        <h1
          className="airport-summary__name"
        >
          Chongqing
        </h1>
        <span
          className="airport-summary__code"
        >
          CKG
        </span>
        <span
          className="airport-summary__location"
        >
          Chonqing
          , 
          China
           -- 
          Asia
        </span>
      </a>
    </div>
  </li>
</ul>
`;
